File: src/area_check/areaCheck.cpp
#include <iostream>
#include <fstream>

int main() {
    // Variables to store latitude and longitude from txt file
    double min_lat, max_lat, min_lon, max_lon;

    // Load file and read
    std::ifstream infile("m_bounds.txt");
    if (!infile.is_open()) {
        std::cerr << "[ERROR] File boundary tidak ditemukan";
        return 1;
    }
    infile >> min_lat >> max_lat >> min_lon >> max_lon;
    infile.close();
    std::cout << "[SUCCESS] Boundary tersimpan ke memori";

    // Simulate AUV reading
    double cur_lat, cur_lon;
    std::cout << "\nEnter Current Latitude: ";
    std::cin >> cur_lat;
    std::cout << "\nEnter Current Longitude: ";
    std::cin >> cur_lon;

    // Check boundary condition
    bool in_lat = (cur_lat >= min_lat && cur_lat <= max_lat);
    bool in_lon = (cur_lon >= min_lon && cur_lon <= max_lon);
    if (in_lat && in_lon) {
        std::cout << "\n[STATUS] Wahana berada dalam zona target";
        std::cout << "[SYSTEM] Menyalakan sistem pemantauan";
    } else {
        std::cout << "\n[STATUS] Wahana berada di luar zona target";
        std::cout << "[SYSTEM] Mematikan sistem pemantauan";
    }

    return 0;
}

File: src/main_detection/areaCheck.py
import json

# Load boundary coordinates from JSON config file
def load_boundaries(filename="m_bounds.json"):
    try:
        with open(filename, "r") as infile:
            bounds = json.load(infile)
            print(f"[SYSTEM] Boundaries loaded from {filename}")
            return bounds
    except FileNotFoundError:
        print(f"[ERROR] File {filename} is not found")
        return None
    except json.JSONDecodeError:
        print(f"[ERROR] File {filename} is corrupted")
        return None

# Check if AUV is inside area
def is_in_area(cur_lat, cur_lon, bounds):
    if bounds is None:
        return False
    
    in_lat = bounds["min_lat"] <= cur_lat <= bounds["max_lat"]
    in_lon = bounds["min_lon"] <= cur_lon <= bounds["max_lon"]

    return in_lat and in_lon # True if coordinate is inside bounding box